tools/scripts/examples_compile.py

```diff
@@ -35,6 +36,9 @@ def generate(project):
 	path = project.parent
 	output = ["=" * 90, "Generating: {}".format(path)]
 	options = " ".join("-D{}={}".format(k, v) for k,v in global_options.items())
+	# Compile Linux examples under macOS with hosted-darwin target
+	if "Darwin" in platform.platform() and "hosted-linux" in project.read_text():
+		options += " -D:target=hosted-darwin"
```

Comment:
@strongly-typed with these changes you should just be able to compile the linux and generic examples without new macOS specific examples.

Reply:
I've added the TravisCI configuration to this PR, lets see how that works.

Reply:
I'm having issues linking with `zmqpp` on both my local system and the CI for `linux/xpcc` and `zmq/*` examples. Can you reproduce this?

Reply:
Moved to a separate PR, since I cannot get all examples to compile.